spec/motion-spec/mock_spec.rb

```diff
@@ -0,0 +1,220 @@
+      @dog.mock!(:eat, return: 'Yuck!')
+      @dog.bark.should == 'Meow!'
+      @dog.eat('broccoli').should == 'Yuck!'
+    end
+
+    it 'should mock using block' do
+      @dog.mock!(:bark) do |a|
+        a.should == 'Meow!'
+        a
+      end
+      @dog.bark('Meow!').should == 'Meow!'
+    end
+
+    it 'should mock a class method' do
+      Dog.mock!(:create, return: 'Dog')
+      Dog.create.should == 'Dog'
+    end
+
+    it 'should be able to yield a single object' do
+      @dog.mock!(:get_toy, yield: 'stuffed fox')
+      @dog.get_toy do |toy|
+        toy.should.be == 'stuffed fox'
+      end
+    end
+
+    it 'should be able to yield multiple objects' do
+      @dog.mock!(:fetch, yield: [false, 10])
+      @dog.fetch do |success, time|
+        success.should.be == success
+        time.should.be == 10
+      end
+    end
+
+    it 'should be able to yield the boolean value false' do
+      @dog.mock!(:get_toy, yield: false)
+      @dog.get_toy do |toy|
+        toy.should.be == false
+      end
+    end
+  end
+
+  describe '#mock' do
+    it 'should create pure mock' do
+      my_mock = mock(:hello, return: 'hi')
+      my_mock.hello.should == 'hi'
+    end
+  end
+
+  describe '#should_not_call' do
+    it 'should raise an error with an instance method' do
+      @dog.should_not_call(:bark)
+      should.raise(MotionSpec::Error) do
+        @dog.bark
+      end
+    end
+
+    it 'should succeed if the call is not made' do
+      should.not.raise(MotionSpec::Error) do
+        @dog.should_not_call(:bark)
+      end
+    end
+
+    it 'should not fail in another test' do
+      should.not.raise(MotionSpec::Error) do
+        @dog.bark
+      end
+    end
+
+    it 'should raise an error with a class method' do
+      Dog.should_not_call(:create)
+      should.raise(MotionSpec::Error) do
+        Dog.create
+      end
+    end
+  end
+
+  describe '#reset' do
+    describe 'stubbing' do
+      it 'should restore original class method' do
+        Dog.stub!(:kind, return: 'Reptile')
+        Dog.kind.should == 'Reptile'
+        Dog.reset(:kind)
+        Dog.kind.should == 'Mammal'
+      end
+
+      it 'should restore original instance method' do
+        @dog.stub!(:bark, return: 'Meow!')
+        @dog.bark.should == 'Meow!'
+        @dog.reset(:bark)
+        @dog.bark.should == 'Woof!'
+      end
+    end
+
+    describe 'mocking' do
+      it 'should restore original class method' do
+        Dog.mock!(:kind, return: 'Reptile')
+        Dog.kind.should == 'Reptile'
+        Dog.reset(:kind)
+        Dog.kind.should == 'Mammal'
+      end
+
+      it 'should restore original instance method' do
+        @dog.mock!(:bark, return: 'Meow!')
+        @dog.bark.should == 'Meow!'
+        @dog.reset(:bark)
+        @dog.bark.should == 'Woof!'
+      end
+    end
+  end
+
+  describe 'after each scenario' do
+    it 'should verify mocks' do
+      MotionSpec::Should.class_eval do
+        def ignored_flunk(message)
```

Comment:
Unused method argument - `message`. If it's necessary, use `_` or `_message` as an argument name to indicate that it won't be used. You can also write as `ignored_flunk(*)` if you want the method to accept any arguments but don't care about them.